recipe/meta.yaml

```diff
@@ -1,63 +1,89 @@
-{% set version = "3.0.0" %}
+{% set version = "3.4.0" %}
 
 package:
   name: freeglut
   version: {{ version }}
 
 source:
-  fn: freeglut-{{ version }}.tar.bz2
-  url: http://sourceforge.net/projects/freeglut/files/freeglut/{{ version }}/freeglut-{{ version }}.tar.gz
-  md5: 90c3ca4dd9d51cf32276bc5344ec9754
+  url: https://sourceforge.net/projects/freeglut/files/freeglut/{{ version }}/freeglut-{{ version }}.tar.gz
+  sha256: 3c0bcb915d9b180a97edaebd011b7a1de54583a838644dcd42bb0ea0c6f3eaec
+  patches:
+    - disable_autolink_windows.patch
 
 build:
-  number: 5
-  # https://github.com/dcnieho/FreeGLUT/issues/27
-  # Maybe someday we'll build with X11 on osx:
-  # https://github.com/Homebrew/homebrew-core/blob/master/Formula/freeglut.rb
-  skip: true                      # [osx]
+  number: 0
+  # mesa-libgl-devel isn't available on s390x
+  skip: true  # [osx or (linux and s390x)]
+  run_exports:
+    # https://abi-laboratory.pro/index.php?view=timeline&l=freeglut
+    # very good backwards compatibility
+    - {{ pin_subpackage("freeglut", max_pin="x") }}
 
 requirements:
   build:
-    - cmake
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
-    - {{ cdt('libxcb') }}               # [linux]
-    - {{ cdt('libxi-devel') }}          # [linux]
+    - cmake
+    - ninja-base
+    - pkg-config
+    - m2-patch                          # [win]
+    - patch                             # [not win]
+    - {{ cdt('libdrm-devel') }}          # [linux]
```

Comment:
A note on why the CDT are required would be nice, no need to go into detail on each one, just a general reason (are they to support libglu and libxcb)?
Also if they're used by libglu and libxcb should they be in host? (not sure, maybe cdts always go in build for some reason)

Reply:
I added comments